pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -21,10 +37,19 @@ def fix_trailing_whitespace(argv=None):
         'grep', '-l', '[[:space:]]$', *args.filenames, retcode=None
     ).strip().splitlines()
 
+    # combine all extension arguments, splitting at ',' and normalizing them
+    # (lowercase and remove unnecessary leading '.' which may be present)
+    md_args = vars(args).get('markdown_linebreak_ext', ['md', 'markdown'])
+    md_exts = [x.lower().lstrip('.') for x in ','.join(md_args).split(',')]
+    all_markdown = '*' in md_exts
+
     if bad_whitespace_files:
         for bad_whitespace_file in bad_whitespace_files:
+            # get extension ([1]); remove leading '.' ([1:]); and make lowercase
+            extension = os.path.splitext(bad_whitespace_file)[1][1:].lower()
```

Comment:
One of the reasons I added the normalization was to be forgiving in the case where somebody uses `--markdown-linebreak-ext=.foo` (rather than `=foo`).  Adding a `.` only if there is not one already is trickier than removing a `.` if present (lstrip), and I certainly don't want to require `.` in the option argument as `--markdown-linebreak-ext=.foo,.bar` is pretty ugly.

It's true that the current code is not self-documenting, but that is why I wrote a comment.